src/main/java/jenkins/bouncycastle/api/PEMEncodable.java

```diff
@@ -0,0 +1,421 @@
+import org.apache.commons.io.FileUtils;
+import org.bouncycastle.openssl.PEMReader;
+import org.bouncycastle.openssl.PEMWriter;
+import org.bouncycastle.openssl.PasswordException;
+import org.bouncycastle.openssl.PasswordFinder;
+import org.bouncycastle.util.encoders.Base64;
+
+/**
+ * A class that provides an API to manager PEM format, providing additional methods to handle Keys, Certificates,
+ * Fingerprints, etc The supported algorithms will depend on the underlying version of BouncyCastle
+ *
+ * @since 1.0
+ */
+public final class PEMEncodable {
+
+    /**
+     * Stores the internal Bouncy Castle or JCA object
+     */
+    @Nonnull
+    private Object object;
+
+    private PEMEncodable(@Nonnull Object pemObject) {
+        this.object = pemObject;
+    }
+
+    /**
+     * Creates a {@link PEMEncodable} from a {@link Key} object
+     * 
+     * @param key {@link Key} object with the key
+     * @return {@link PEMEncodable} object
+     */
+    @Nonnull
+    public static PEMEncodable create(@Nonnull Key key) {
+        return new PEMEncodable(key);
+    }
+
+    /**
+     * Creates a {@link PEMEncodable} from a {@link KeyPair} object
+     * 
+     * @param keyPair {@link KeyPair} object with the key pair
+     * @return {@link PEMEncodable} object
+     */
+    @Nonnull
+    public static PEMEncodable create(@Nonnull KeyPair keyPair) {
+        return new PEMEncodable(keyPair);
+    }
+
+    /**
+     * Creates a {@link PEMEncodable} from a {@link Certificate} object
+     * 
+     * @param certificate {@link Certificate} object with the certificate
+     * @return {@link PEMEncodable} object
+     */
+    @Nonnull
+    public static PEMEncodable create(@Nonnull Certificate certificate) {
+        return new PEMEncodable(certificate);
+    }
+
+    /**
+     * Creates a {@link PEMEncodable} by decoding PEM formated data from a {@link String}
+     * 
+     * @param pem {@link String} with the PEM data
+     * @return {@link PEMEncodable} object
+     * @throws IOException launched if a problem exists reading the PEM information
+     * @throws UnrecoverableKeyException in case PEM is passphrase protected and none or wrong is provided
+     */
+    @Nonnull
+    public static PEMEncodable decode(@Nonnull String pem) throws IOException, UnrecoverableKeyException {
+        return decode(pem, null);
+    }
+
+    /**
+     * Creates a {@link PEMEncodable} by decoding PEM formated data from a {@link String}
+     * 
+     * @param pem {@link String} with the PEM data
+     * @param passphrase passphrase for the encrypted PEM data. null if PEM data is not passphrase protected
+     * @return {@link PEMEncodable} object
+     * @throws IOException launched if a problem exists reading the PEM information
+     * @throws UnrecoverableKeyException in case PEM is passphrase protected and none or wrong is provided
+     */
+    @Nonnull
+    public static PEMEncodable decode(@Nonnull String pem, @Nullable final String passphrase)
```

Comment:
So this signature exposes a bad practice. When you store the passphrase in a `String` you have no control over the lifetime of the `String` in memory. Best practice is to pass the passphrase in a `char[]` and document in the Javadoc that the caller is responsible for zeroing out the `char[]` after use, e.g. with `Arrays.fill(passphrase, (char)0)`

By replicating this bad practice we do not enhance the security of consumers of this API